go/vt/schema/ddl_strategy_test.go

```diff
@@ -22,21 +22,40 @@ import (
 	"time"
 
 	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	tabletmanagerdatapb "vitess.io/vitess/go/vt/proto/tabletmanagerdata"
 )
 
 func TestIsDirect(t *testing.T) {
-	assert.True(t, DDLStrategyDirect.IsDirect())
-	assert.False(t, DDLStrategyVitess.IsDirect())
-	assert.False(t, DDLStrategyOnline.IsDirect())
-	assert.False(t, DDLStrategyGhost.IsDirect())
-	assert.False(t, DDLStrategyPTOSC.IsDirect())
-	assert.True(t, DDLStrategy("").IsDirect())
-	assert.False(t, DDLStrategy("vitess").IsDirect())
-	assert.False(t, DDLStrategy("online").IsDirect())
-	assert.False(t, DDLStrategy("gh-ost").IsDirect())
-	assert.False(t, DDLStrategy("pt-osc").IsDirect())
-	assert.False(t, DDLStrategy("mysql").IsDirect())
-	assert.True(t, DDLStrategy("something").IsDirect())
+	assert.True(t, NewDDLStrategySetting(tabletmanagerdatapb.OnlineDDL_DIRECT, "").IsDirect())
+	assert.False(t, NewDDLStrategySetting(tabletmanagerdatapb.OnlineDDL_VITESS, "").IsDirect())
+	assert.False(t, NewDDLStrategySetting(tabletmanagerdatapb.OnlineDDL_ONLINE, "").IsDirect())
+	assert.False(t, NewDDLStrategySetting(tabletmanagerdatapb.OnlineDDL_GHOST, "").IsDirect())
+	assert.False(t, NewDDLStrategySetting(tabletmanagerdatapb.OnlineDDL_PTOSC, "").IsDirect())
+	assert.False(t, NewDDLStrategySetting(tabletmanagerdatapb.OnlineDDL_MYSQL, "").IsDirect())
+
+	for name, direct := range map[string]bool{
+		"":       true,
+		"vitess": false,
+		"online": false,
+		"gh-ost": false,
+		"pt-osc": false,
+		"mysql":  false,
+	} {
+		strategy, err := ParseDDLStrategyName(name)
+		require.NoError(t, err)
+
+		var assertion func(t assert.TestingT, value bool, msgAndArgs ...any) bool
+		switch direct {
+		case true:
+			assertion = assert.True
+		case false:
+			assertion = assert.False
+		}
+
+		assertion(t, NewDDLStrategySetting(strategy, "").IsDirect())
```

Comment:
I think this reads simpler:
```suggestion
    assert.Equal(t, direct, NewDDLStrategySetting(strategy, "").IsDirect())
```